packages/runtime/src/util/validate-plan.ts

```diff
@@ -20,6 +24,54 @@ export default (plan: ExecutionPlan) => {
   return true;
 };
 
+const assertWorkflowStructure = (plan: ExecutionPlan) => {
+  const { workflow, options } = plan;
+
+  if (!workflow || typeof workflow !== 'object') {
+    throw new ValidationError('Missing or invalid workflow key in execution plan.');
+  }
+
+  if (!Array.isArray(workflow.steps)) {
+    throw new ValidationError('The workflow.steps key must be an array.');
+  }
+
+  if (workflow.steps.length === 0) {
+    logger.warn('Warning: The workflow.steps array is empty.');
+  }
+
+  workflow.steps.forEach((step, index) => {
+    assertStepStructure(step, index);
+  });
+
+  if (options) {
+    assertOptionsStructure(options);
```

Comment:
You can remove this if statement if you default `options` to an empty object inside `assertOptionStructure`, resulting in more elegant code :)

```
const assertOptionsStructure = (options: WorkflowOptions = {}) => { ... }
```